pkg/table/tables/partition.go

```diff
@@ -960,7 +966,7 @@ func (lp *ForListPruning) buildListColumnsPruner(ctx expression.BuildContext,
 // colIdx is the column index in the list columns.
 func (lp *ForListPruning) buildListPartitionValueMap(ctx expression.BuildContext, defs []model.PartitionDefinition,
 	schema *expression.Schema, names types.NameSlice, p *parser.Parser) error {
-	lp.valueMap = map[int64]int{}
+	lp.valueMap = btree.NewG[*btreeListItem](btreeDegree, func(a, b *btreeListItem) bool { return a.key < b.key })
```

Comment:
Are there any tests for `bigint unsigned` List partitioning columns? And how `bigint signed` handles comparison between MaxInt64 and MaxUInt64?

Reply:
Good point, I will fix it and add some test cases.